### PR TITLE
[nexus] Add tests for instance_create unwind safety

### PR DESCRIPTION
**Pull request by smklein**
Builds on https://github.com/oxidecomputer/omicron/pull/1835 and https://github.com/oxidecomputer/steno/pull/67

Adds a test for the "instance create" saga's unwind safety by:
- Using a saga which attempts to create a network interface, allocate an external IP, and attach a disk
- Failing after *every single intermediate node* in the saga
- Verifying that no intermediate resources remain after any of the attempted executions

### CODE REVIEW

File: common/Cargo.toml [hunk 1]
```diff
@@ -25,7 +25,7 @@ serde_json = "1.0"
 serde_with = "2.0.1"
 slog = { version = "2.5", features = [ "max_level_trace", "release_max_level_debug" ] }
 smf = "0.2"
-steno = "0.2"
+steno = { git = "https://github.com/oxidecomputer/steno", branch = "node-access" }
```
Comment (smklein) [inline common/Cargo.toml:28]: Ideally, I'd use a new rev of steno instead before merging. Same elsewhere in this PR

File: nexus/src/app/sagas/instance_create.rs [hunk 2]
```diff
@@ -1014,3 +1014,234 @@ async fn sic_instance_ensure(
...
+        let nexus = &cptestctx.server.apictx.nexus;
+        let project_id = create_org_project_and_disk(&client).await;
+
+        // Build the saga DAG with the provided test parameters
+        let opctx = cptestctx.test_opctx();
+
+        let params = new_test_params(&opctx, project_id);
+        let dag = create_saga_dag::<SagaInstanceCreate>(params).unwrap();
+
+        for i in 0..dag.get_node_count() {
```
Comment (smklein) [inline nexus/src/app/sagas/instance_create.rs:1204]: This is the "more automated" version of what I tried doing in https://github.com/oxidecomputer/omicron/pull/1835 - I can definitely consolidate the testing here, and create a "saga unwind tester", if that would be useful.

I basically see three phases:
1. Setup the saga. Create any orgs / projects / disks / etc needed to run the saga.
2. Iterate over every single node in the DAG, injecting failures at each one. Log which ones we're working on, to make it clear what went wrong if an error occurs. Run the saga.
3. Assert that no detritus remains, and that the system is in the same state before / after saga execution.

**Review by davepacheco — APPROVED** [review]
Cool!

**Review by leftwo — APPROVED** [review]
Yeah, this looks great.  I look forward to this going back so I can do the same things
with the volume sagas.

**Pull request merged**